backend/src/wamp.js

```diff
@@ -101,6 +103,8 @@ function setupWamp() {
         return;
       }
     }
+
+    wamp_session = session;
```

Comment:
[According to the source code](https://github.com/crossbario/autobahn-js/blob/9097e868e2bc46e599b1c16e6ba25c3dc14bb644/packages/autobahn/lib/connection.js#L416-L420), `wamp` instance should have `.session` property, so you don't need to save it separately, just use `wamp.session`

Reply:
I tried, `wamp.session` is null

Reply:
There is something wrong with your experiment since [according to the source code](https://github.com/crossbario/autobahn-js/blob/9097e868e2bc46e599b1c16e6ba25c3dc14bb644/packages/autobahn/lib/connection.js#L292) `onopen` is called with the same self._session as it returns from `wamp.session`. `wamp.session` will be null when it is not connected or was disconnected, in your case you keep even dead `wamp_session`, which may lead to many problems in the future. Use `wamp.session` and let me know if you still cannot get it working.